bluesky_queueserver/manager/manager.py

```diff
@@ -0,0 +1,655 @@
+            await self._clear_running_plan_info()
+
+    # ======================================================================
+    #   Functions that implement functionality of the server
+
+    async def _start_re_worker(self):
+        """
+        Creates worker process.
+        """
+        self._event_worker_created = asyncio.Event()
+
+        msg = {"type": "command", "value": "start_re_worker"}
+        self._watchdog_conn.send(msg)
+
+        logger.debug("Waiting for RE worker to start ...")
+        await self._event_worker_created.wait()
+        logger.debug("Worker started successfully.")
+
+    async def _started_re_worker(self):
+        # Report from RE Worker received: environment was created successfully.
+        self._event_worker_created.set()
+
+    async def _stop_re_worker(self):
+        """
+        Closes Run Engine execution environment in orderly way. Running plan needs
+        to be stopped before the environment can be closed.
+        """
+        if self._environment_exists:
+            self._event_worker_closed = asyncio.Event()
+
+            msg = {"type": "command", "value": "quit"}
+            self._worker_conn.send(msg)
+
+            logger.debug("Waiting for RE Worker to close ...")
+            await self._event_worker_closed.wait()
+            logger.debug("RE Worker to closed")
+
+            msg = {"type": "command", "value": "join_re_worker"}
+            self._watchdog_conn.send(msg)
+
+            self._environment_exists = False
+            return True
+        else:
+            return False
+
+    async def _stopped_re_worker(self):
+        # Report from RE Worker received: environment was closed successfully.
+        self._event_worker_closed.set()
+
+    async def _is_worker_alive(self):
+        self._fut_is_worker_alive = self._loop.create_future()
+
+        msg = {"type": "request", "value": "is_worker_alive"}
+        self._watchdog_conn.send(msg)
+
+        return await self._fut_is_worker_alive
+
+    async def _is_worker_alive_received(self, is_alive):
+        self._fut_is_worker_alive.set_result(is_alive)
+
+    async def _worker_status_request(self):
+        self._fut_worker_status = self._loop.create_future()
+
+        msg = {"type": "request", "value": "status"}
+        self._worker_conn.send(msg)
+
+        return await self._fut_worker_status
+
+    async def _worker_status_received(self, status):
+        self._fut_worker_status.set_result(status)
+
+    async def _run_task(self):
+        """
+        Upload the plan to the worker process for execution.
+        Plan in the queue is represented as a dictionary with the keys "name" (plan name),
+        "args" (list of args), "kwargs" (list of kwargs). Only the plan name is mandatory.
+        Names of plans and devices are strings.
+        """
+        n_pending_plans = await self._r_pool.llen('plan_queue')
+        logger.info("Starting a new plan: %d plans are left in the queue" % n_pending_plans)
```

Comment:
```suggestion
        logger.info("Starting a new plan: %d plans are left in the queue", n_pending_plans)
```

This defers doing the templating until the loggers need it.

Reply:
I see, I will modify the logging commands in the next PR.